crates/bevy_ui/src/geometry.rs

```diff
@@ -0,0 +1,361 @@
+use bevy_math::Vec2;
+use bevy_reflect::Reflect;
+use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign};
+
+/// A 2-dimensional area defined by a width and height.
+///
+/// It is commonly used to define the size of a text or UI element.
+///
+/// # Example
+///
+/// In this example an UI size gets created. It has a width of 100px and a height of 200px.
+///
+/// ```rust
+/// # use bevy_ui::{Size, Val};
+/// #
+/// let size = Size {
+///     width: Val::Px(100.0),
+///     height: Val::Px(200.0),
+/// };
+///
+/// assert_eq!(size.width, Val::Px(100.0));
+/// assert_eq!(size.height, Val::Px(200.0));
+/// ```
+#[derive(Copy, Clone, PartialEq, Debug, Reflect)]
+#[reflect(PartialEq)]
+pub struct Size<T: Reflect + PartialEq = f32> {
+    /// The width of the 2-dimensional area.
+    pub width: T,
+    /// The height of the 2-dimensional area.
+    pub height: T,
+}
+
+impl<T: Reflect + PartialEq> Size<T> {
+    /// Creates a new [`Size`] from a width and a height.
+    ///
+    /// # Example
+    ///
+    /// In this example an UI size gets created. It has a width of 100px and a height of 200px.
+    ///
+    /// ```rust
+    /// # use bevy_ui::{Size, Val};
+    /// #
+    /// let size = Size::new(Val::Px(100.0), Val::Px(200.0));
+    ///
+    /// assert_eq!(size.width, Val::Px(100.0));
+    /// assert_eq!(size.height, Val::Px(200.0));
+    /// ```
+    pub fn new(width: T, height: T) -> Self {
+        Size { width, height }
+    }
+}
+
+impl<T: Default + Reflect + PartialEq> Default for Size<T> {
+    fn default() -> Self {
+        Self {
+            width: Default::default(),
+            height: Default::default(),
+        }
+    }
+}
+
+/// A type which is commonly used to define positions, margins, paddings and borders.
+///
+/// # Examples
+///
+/// ## Position
+///
+/// A position is used to determine where to place an UI element.
```

Comment:
```suggestion
/// A position is used to determine where to place a UI element.
```

English wants "a UI" and "a user interface". I know it starts with a vowel, but the "u" is hard, and uses a "y" sound. There's a few places where this happens if you search <3

Reply:
Good to know. Thanks! I'll change these to `a`.